homeassistant/components/netatmo/sensor.py

```diff
@@ -73,23 +74,54 @@
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the available Netatmo weather sensors."""
     netatmo = hass.components.netatmo
-    data = NetAtmoData(netatmo.NETATMO_AUTH, config.get(CONF_STATION, None))
 
     dev = []
+    if CONF_MODULES in config:
+        manual_config(netatmo, config, dev)
+    else:
+        auto_config(netatmo, config, dev)
+
+    if dev:
+        add_entities(dev, True)
+
+
+def manual_config(netatmo, config, dev):
+    """Handle manual configuration."""
     import pyatmo
-    try:
-        if CONF_MODULES in config:
+
+    not_handled = []
+    for data_class in [pyatmo.WeatherStationData]:
```

Comment:
No `HomeCoachData` here?

Reply:
No not changed how the manual config is handled. Works just as before.

Reply:
Hmm. You moved the class from `_detect_platform_type` to `auto_config` so it seems like, for `manual_config`, it was checked before but now it isn't.

(I may be reading this wrong, I do not have devices to test the PR)

Reply:
Yes you are right. Fixed. I have tested the different combinations in manual setup and it all works now.